src/Docker.DotNet/Endpoints/ImageOperations.cs

```diff
@@ -255,9 +255,7 @@ public async Task<CommitContainerChangesResponse> CommitContainerChangesAsync(Co
                 throw new ArgumentNullException(nameof(parameters));
             }
 
-            var data = parameters.Config == null
-                ? null
-                : new JsonRequestContent<Config>(parameters.Config, this._client.JsonSerializer);
+            var data = new JsonRequestContent<Config>(parameters.Config ?? new Config(), this._client.JsonSerializer);
```

Comment:
I do not believe this is the appropriate fix. As I mentioned in the related issue, we should generate the class `CommitContainerChangesParameters` (via [specgen](https://github.com/dotnet/Docker.DotNet/tree/master/tools/specgen)) similar to [CreateContainerParameters](https://github.com/dotnet/Docker.DotNet/blob/a0eba61f7eb0ebec1e52f38c3b91c28b54128a6f/src/Docker.DotNet/Models/CreateContainerParameters.Generated.cs) and remove the nested type. Then we can simply call:

```suggestion
            var data = new JsonRequestContent<CommitContainerChangesParameters>(parameters, this._client.JsonSerializer);
```

Reply:
I have no updated modeldefs and regenerated the classes.